lib/rbac/filterer.rb

```diff
@@ -417,17 +402,33 @@ def scope_to_tenant(scope, user, miq_group)
       tenant_id_clause ? scope.where(tenant_id_clause) : scope
     end
 
+    ##
+    # Main scoping method
+    #
     def scope_targets(klass, scope, rbac_filters, user, miq_group)
+      # Results are scoped by tenant if the TenancyMixin is included in the class,
+      # with a few manual exceptions (User, Tenant). Note that the classes in
+      # TENANT_ACCESS_STRATEGY are a consolidated list of them.
       if klass.respond_to?(:scope_by_tenant?) && klass.scope_by_tenant?
```

Comment:
I'd prefer to use rbac lookup tables over methods.
OTOH, I'd prefer to use methods over `klass < ParentClass`

Also if possible, the lookup tables are driven by input into Rbac.

This will allow us to generate scopes easily and get rid of this logic.

Going a direction like `scope_by_tenant?` looks better than `CLASSES_WITH_TENANCY` until `scope_by_tenant?` starts getting overridden by children. Then we get into trouble.